### ISSUE TITLE
Is there a way to map to pyarrow.hdfs.connect?

### ISSUE THREAD

**Issue by Jeffwan** [fsspec/filesystem_spec#1113]
Let's say we have a file system called `bfs://` which is an equivalent implementation of HDFS. We can support pyarrow operation like following way.

```
import pyarrow as pa
from pyarrow import fs
bfs = pa.hdfs.connect('bfs://service-endpoint')
local, _ = fs.FileSystem.from_uri('file:///')
with local.open_input_stream('/tmp/license.txt') as src:
    with bfs.open('/tmp-test.txt', mode='wb') as dest:
        dest.upload(src)
```

It doesn't support below way yet and HadoopFileSystem implementation seems doesn't work. 

```
from pyarrow import fs
fs.FileSystem.from_uri('bfs://service-endpoint')
```

Could I know whether fsspec can support our protocol?

**Comment by martindurant** [comment]
Could you please explain how you'd like your alternative implementation to be handled by fsspec? I am understanding, that you would like the "bfs" protocol to be registered with fsspec, and have it create the arrow fs, wrap it, and return an fsspec instance. Is this right?

**Comment by Jeffwan** [comment]
@martindurant Yes. That's exact what I want. Do you have any suggestion for those custom protocols? Should we do it downstream or upstream?

**Comment by martindurant** [comment]
@Jeffwan , did my proposed solution work for you?
